spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/publisher/PubSubPublisherTemplate.java

```diff
@@ -103,7 +106,10 @@ public void setMessageConverter(PubSubMessageConverter pubSubMessageConverter) {
 			@Override
 			public void onFailure(Throwable throwable) {
 				LOGGER.warn("Publishing to " + topic + " topic failed.", throwable);
-				settableFuture.setException(throwable);
+				Message<?> originalMessage = MessageBuilder.withPayload(pubsubMessage).build();
+				MessageHandlingException messageHandlingException = new MessageHandlingException(
```

Comment:
Can you use `MessageDeliveryException` instead please?

Reply:
Well, I wouldn't do that.
I mean we are good to have a `PubSubDeliveryException` as an extension of the `PubSubException`.
My point is to not wrap a `PubsubMessage` into a `Message<?>`: an original request is not Spring Messaging.
So, let's not abuse non-related API, but we definitely can borrow an idea! 😄

Reply:
Thank you for your comments. I have updated the code accordingly. Can you please check the updated commit.